Source/WebKit/WebProcess/WebPage/WebCookieJar.cpp

```diff
@@ -310,4 +329,18 @@ void WebCookieJar::setCookieAsync(WebCore::Document& document, const URL& url, c
     WebProcess::singleton().ensureNetworkProcessConnection().connection().sendWithAsyncReply(Messages::NetworkConnectionToWebProcess::SetCookieFromDOMAsync(document.firstPartyForCookies(), sameSiteInfo, url, frameID, pageID, applyTrackingPreventionInNetworkProcess, shouldRelaxThirdPartyCookieBlocking(webFrame), cookie), WTFMove(completionHandler));
 }
 
+#if HAVE(COOKIE_CHANGE_LISTENER_API)
+void WebCookieJar::addCookieChangeListener(const String& host, const WebCore::CookieChangeListener& cookieChangeListener)
+{
+    WebProcess::singleton().ensureNetworkProcessConnection().connection().send(Messages::NetworkConnectionToWebProcess::SubscribeToCookieChangeNotifications(host), 0, IPC::SendOption::DispatchMessageEvenWhenWaitingForSyncReply);
+    m_cookieChangeListeners.add(host, WeakPtr { cookieChangeListener });
+}
+
+void WebCookieJar::removeCookieChangeListener(const String& host)
+{
+    WebProcess::singleton().ensureNetworkProcessConnection().connection().send(Messages::NetworkConnectionToWebProcess::UnsubscribeFromCookieChangeNotifications(HashSet<String> { host }), 0, IPC::SendOption::DispatchMessageEvenWhenWaitingForSyncReply);
+    m_cookieChangeListeners.remove(host);
```

Comment:
That's wrong, there can be several listeners for the same host. You need to pass the listener as parameter.